### ISSUE TITLE
[enhancement]: Add monaco editor to the prompt and negative prompt fields

### ISSUE THREAD

**Issue by airtonix** [invoke-ai/InvokeAI#2859]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What should this feature add?

If you add monaco editor (the editor under vscode) to anywhere we do prompts, then we'll be able to do: 

- automatic surrounds (highlight a word/phrase/sentence, press `(` and have it wrap the selection)
- syntax highlighting for the `(`, `[`, `(.*[\d\.\d])` etc
- invalid syntax highlighting for incorrectly supported syntax: `<lora etc etc>`
- auto complete available textual inversion keywords

I'm sure the possibilities are endless. but the above are some nice low hanging fruit.

### Alternatives

sell my computer and become a farmer? 

### Aditional Content

- the editor: https://microsoft.github.io/monaco-editor/
- syntax highlighting: https://microsoft.github.io/monaco-editor/monarch.html
- https://codesandbox.io/s/monaco-editor-react-u0fyv

**Comment by simonfuhrmann** [comment]
This is quite a heavy weight editor for marginally better editing capabilities.

**Comment by airtonix** [comment]
> This is quite a heavy weight editor for marginally better editing capabilities.

Not really. You'll find it in use pretty much anywhere entering text is more than a sms message.
